sdk/core/Azure.Core/src/Shared/AzureKeyCredentialPolicy.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    internal class AzureKeyCredentialPolicy : HttpPipelineSynchronousPolicy
+    {
+        private readonly string _keyHeader;
+        private readonly AzureKeyCredential _credential;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureKeyCredentialPolicy"/> class.
+        /// </summary>
+        /// <param name="credential">The <see cref="AzureKeyCredential"/> used to authenticate requests.</param>
+        /// <param name="keyHeader">The name of the API key header used for signing requests.</param>
+        public AzureKeyCredentialPolicy(AzureKeyCredential credential, string keyHeader)
+        {
+            Argument.AssertNotNull(credential, nameof(credential));
+            Argument.AssertNotNullOrEmpty(keyHeader, nameof(keyHeader));
+            _credential = credential;
+            _keyHeader = keyHeader;
+        }
+
+        /// <inheritdoc/>
+        public override void OnSendingRequest(HttpMessage message)
+        {
+            base.OnSendingRequest(message);
+            message.Request.Headers.SetValue(_keyHeader, _credential.ApiKey);
```

Comment:
@schaabs, are there cases where the key needs to be somehow transformed before it ends up in the header or it's written to the header as-is?

Reply:
There are cases where the key is not directly inserted as a header value, but rather used to sign a request. However this policy is specifically for cases where the key is simply added as a header. Perhaps a more specific type name would be better, but given that this type is internal it would be for library authors benefit.